crates/server/src/client.rs

```diff
@@ -469,6 +472,13 @@ impl Client {
     fn send_packet(&self, packet: impl Into<ServerPlayPacket>) {
         let _ = self.packets_to_send.try_send(packet.into());
     }
+
+    pub fn disconnect(&self, reason: &str) {
+        self.disconnected.set(true);
+        self.send_packet(Disconnect {
```

Comment:
Could client not just intercept all disconnect packets?

Reply:
This isn't really part of this PR, and setting disconnected kills the tcp stream.